### ISSUE TITLE
JAWS tablist functionality does not work with tab component

### ISSUE THREAD

**Issue by abbott567** [alphagov/govuk-frontend#4735]
## Description of the issue
There is a bug with the tab component when you use the tablist feature of JAWS.

The tablist feature allows JAWS users to pull all of the tabs on the page in one list and activate them from there, rather than navigating to them on the page itself via the <kbd>TAB</kbd> and arrow keys.

When you view the tabs in the tablist, and you choose one, the tabslist closes and the focus is moved to the tab you chose from the list. At this point the tab is not activated the way it would be if you click on it or use the arrow keys to get to it. It has keyboard focus, but it has not been activated. JAWS then incorrectly announces that you can press <kbd>SPACE</kbd> to activate the tab, but you cannot, which is super confusing.

We've tried testing in other screen readers, such as Voiceover using the rotor feature, it behaves similarly in that it moves the focus but does not activate the tab. However, with Voiceover, it tells you specifically to use <kbd>CONTROL</kbd> + <kbd>OPTION</kbd> + <kbd>SPACE</kbd> rather than just <kbd>SPACE</kbd>, which does actually work.

It appears JAWS has certain expectations from a link, even when it's using `role="tab"`, and that is that you can activate it using <kbd>SPACE</kbd>, but the links on the GDS tab component do not activate this way like regular links.

We tested the [Inclusive components - tabbed interfaces example](https://inclusive-components.design/tabbed-interfaces/) and found the same issues, so we can only assume the GDS component has inherited these problems when using it as inspiration. By using the tablist you can actually break the original component to show multiple tab sections at once as it seems to bypass the expected Javascript event triggers which is highlighted in the following screenshot. The GDS component does not seem to have inherited this issue though, thankfully.

<img width="718" alt="Example of multiple tab sections being open at once. Both section 1 and section 4 are open and their tabs highlighted." src="https://user-images.githubusercontent.com/9450064/159286168-3a33b8b4-02fa-4886-9cea-0a45a0d1d78d.png">

We also tested the [example of tabs with automatic activation](https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html) by W3C, and this one seems to behave correctly. When JAWS says you can use <kbd>SPACE</kbd>, you can, and it just works.

## Steps to reproduce the issue
1. Press <kbd>JAWS</kbd> + <kbd>F3</kbd> to bring up the rotor.
2. Navigate to tabslist.
3. Choose a tab from the list
4. Press <kbd>SPACE</kbd> when JAWS tells you to

## Actual vs expected behaviour
When JAWS says press <kbd>SPACE</kbd> to activate the tab, we expect to be able to press <kbd>SPACE</kbd> to activate it. But you cannot. When you press <kbd>SPACE</kbd> nothing happens.

## Environment (where applicable)
- Operating system: Windows 10
- Browser: Microsoft Edge
- Browser version: 96.0.1
- JAWS Version: 2021

**Comment by querkmachine** [comment]
I've given this a re-test in updated versions of Edge and JAWS (Edge 127, JAWS 2023.2303.144) and could no longer reproduce this issue as described. Focusing a tab via the VirtualHTMLFeatures menu and pressing the space bar changed the active tab as intended.

Likewise, I could not reproduce the issue described on the Inclusive Components tabs component.

As we haven't done anything to the component to resolve this, my assumption is that past versions of JAWS handled this in a weird way which has since been resolved.

It'd be good to have this looked at by someone else to verify if this is the case.

**Comment by querkmachine** [comment]
@selfthinker Sorry for the ping, but you might be best placed to check if this bug still exists.

**Comment by selfthinker** [comment]
@querkmachine, just to let you know, this is on my todo list.
I will look at it when I'm back from my leave in two weeks.
